api/metrics/multi_gatherer.go

```diff
@@ -4,92 +4,48 @@
 package metrics
 
 import (
-	"cmp"
-	"errors"
 	"fmt"
-	"slices"
 	"sync"
 
 	"github.com/prometheus/client_golang/prometheus"
 
-	"github.com/ava-labs/avalanchego/utils/metric"
-
 	dto "github.com/prometheus/client_model/go"
 )
 
-var (
-	_ MultiGatherer = (*multiGatherer)(nil)
-
-	errReregisterGatherer = errors.New("attempt to register existing gatherer")
-)
-
 // MultiGatherer extends the Gatherer interface by allowing additional gatherers
 // to be registered.
 type MultiGatherer interface {
 	prometheus.Gatherer
 
 	// Register adds the outputs of [gatherer] to the results of future calls to
-	// Gather with the provided [namespace] added to the metrics.
-	Register(namespace string, gatherer prometheus.Gatherer) error
+	// Gather with the provided [name] added to the metrics.
+	Register(name string, gatherer prometheus.Gatherer) error
 }
 
-type multiGatherer struct {
-	lock      sync.RWMutex
-	gatherers map[string]prometheus.Gatherer
+// Deprecated: Use NewPrefixGatherer instead.
+//
+// TODO: Remove once coreth is updated.
+func NewMultiGatherer() MultiGatherer {
```

Comment:
This is currently used by coreth... Figured keeping it around for now was easier than updating the dependency for now.